lib/rules/nesting-selector-no-missing-scoping-root/index.mjs

```diff
@@ -0,0 +1,159 @@
+import isStandardSyntaxRule from '../../utils/isStandardSyntaxRule.mjs';
+import parseSelector from '../../utils/parseSelector.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'nesting-selector-no-missing-scoping-root';
+
+const messages = ruleMessages(ruleName, {
+	rejected: 'Unexpected missing scoping root',
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/nesting-selector-no-missing-scoping-root',
+};
+
+/** @type {import('stylelint').CoreRules[ruleName]} */
+const rule = (primary) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, {
+			actual: primary,
+			possible: [true],
+		});
+
+		if (!validOptions) return;
+
+		root.walkRules(/&/, (ruleNode) => {
+			if (!isStandardSyntaxRule(ruleNode)) return;
+
+			const selector = getRuleSelector(ruleNode);
+
+			parseSelector(selector, result, ruleNode)?.walkNesting((nestingNode) => {
+				// Check if the rule is nested within a scoping root
+				if (hasValidScopingRoot(ruleNode)) return;
+
+				const { sourceIndex: index } = nestingNode;
+				const endIndex = index + 1;
+
+				report({
+					message: messages.rejected,
+					node: ruleNode,
+					result,
+					ruleName,
+					index,
+					endIndex,
+				});
+			});
+		});
+
+		// Check @scope at-rules for nesting selectors in parameters
+		root.walkAtRules(atRuleRegexes.scopeName, (atRule) => {
+			if (!isStandardSyntaxAtRule(atRule)) return;
+
+			// Only reject orphaned @scope at top level (without parent rules)
+			if (hasValidScopingRoot(atRule)) return;
+
+			const params = getAtRuleParams(atRule);
+
+			if (!params) return;
+
+			checkScopeParams(params, atRule, result);
+		});
+	};
+};
+
+/**
+ * Check if a node has a valid scoping root
+ * @param {import('postcss').Rule | import('postcss').AtRule} node
+ * @returns {boolean}
+ */
+function hasValidScopingRoot(node) {
+	let current = node.parent;
+
+	while (current) {
+		// If we find a rule in the parent chain, it provides a scoping root
+		if (current.type === 'rule') {
+			return true;
+		}
+
+		// If we find an @scope at-rule, it provides a scoping root
+		if (current.type === 'atrule' && current.name === 'scope') {
+			return true;
+		}
+
+		// If we reach the root without finding a scoping root
+		if (current.type === 'root') {
+			return false;
+		}
+
+		current = current.parent;
+	}
+
+	return false;
+}
+
+/**
+ * Check @scope at-rule parameters for nesting selectors
+ * @param {string} params
+ * @param {import('postcss').AtRule} atRule
+ * @param {import('stylelint').PostcssResult} result
+ */
+function checkScopeParams(params, atRule, result) {
+	const parenthesesRegex = /\(([^)]+)\)/g;
+	let match;
+
+	while ((match = parenthesesRegex.exec(params)) !== null) {
+		const selectorText = match[1]?.trim();
```

Comment:
Let's refactor this to use the CSSTree parser to [adhere to our conventions](https://stylelint.io/developer-guide/rules/#construct-specific-parsers).

We prefer the robustness of a parser over regex, as using regex has tripped us up in the past for some edge cases.

Here's an example of [another rule using the parser](https://github.com/stylelint/stylelint/blob/9a4e2b2bc5ced683dcbb67d5eaa62e2e832e489b/lib/rules/selector-anb-no-unmatchable/index.mjs).